### ISSUE TITLE
[HttpClient] query configuration is no longer respected in 5.2.8

### ISSUE THREAD

**Issue by mynameisbogdan** [symfony/symfony#41220]
**Symfony version(s) affected**: 5.2.8

**Description**
Setting a `query` configuration for a scoped client is no longer respected in 5.2.8 

**How to reproduce**
```
framework:
    http_client:
        scoped_clients:
            example_api.http_client:
                base_uri: '%env(EXAMPLE_API_BASE_URI)%'
                query:
                    queryCode: '%env(EXAMPLE_API_QUERY_CODE)%'
```

**Possible Solution**

**Additional context**
Reverting back to 5.2.7 it works as indented, as in `queryCode` is sent in the request.

**Comment by bykclk** [comment]
This has been modified in https://github.com/symfony/symfony/pull/40047. I guess it deliberately deleted it.

**Comment by mynameisbogdan** [comment]
The only recent change is #41160

**Comment by nicolas-grekas** [comment]
Thanks for the report, this is fixed in #41224
